src/Hazelcast.Net/Serialization/Compact/CompactOptions.cs

```diff
@@ -0,0 +1,462 @@
+        /// <param name="schema">The schema.</param>
+        /// <param name="isClusterSchema">Whether the schema is known by the cluster already.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="serializedType"/> or <paramref name="schema"/> is <c>null</c>.</exception>
+        /// <exception cref="ConfigurationException">The operation conflicts with information that were already provided.</exception>
+        internal void SetSchema(Type serializedType, Schema schema, bool isClusterSchema)
+        {
+            if (serializedType == null) throw new ArgumentNullException(nameof(serializedType));
+            if (schema == null) throw new ArgumentNullException(nameof(schema));
+
+            var typeName = schema.TypeName;
+
+            EnsureUniqueTypeNamePerSerializedType(serializedType, typeName);
+            EnsureUniqueSchemaPerTypeName(schema);
+            EnsureTypeNameSerializerCanSerializeType(typeName, serializedType);
+
+            _serializedType.Add(serializedType);
+            _serializedType_typeName[serializedType] = typeName;
+            _typeName_schema[schema.TypeName] = schema;
+
+            _typeName_isClusterSchema[schema.TypeName] = isClusterSchema;
+            _serializedType_isClusterSchema[serializedType] = isClusterSchema;
+        }
+
+        /// <summary>
+        /// Configures a schema for a serialized type.
+        /// </summary>
+        /// <param name="serializedType">The serialized type.</param>
+        /// <param name="isClusterSchema">Whether the schema is known by the cluster already.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="serializedType"/> is <c>null</c>.</exception>
+        internal void SetSchema(Type serializedType, bool isClusterSchema)
+        {
+            if (serializedType == null) throw new ArgumentNullException(nameof(serializedType));
+
+            _serializedType.Add(serializedType);
+            _serializedType_isClusterSchema[serializedType] = isClusterSchema;
+        }
+
+        /// <summary>
+        /// Configures a schema.
+        /// </summary>
+        /// <param name="schema">The schema.</param>
+        /// <param name="isClusterSchema">Whether the schema is known by the cluster already.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="schema"/> is <c>null</c>.</exception>
+        /// <exception cref="ConfigurationException">The operation conflicts with information that were already provided.</exception>
+        internal void SetSchema(Schema schema, bool isClusterSchema)
+        {
+            if (schema == null) throw new ArgumentNullException(nameof(schema));
+
+            EnsureUniqueSchemaPerTypeName(schema);
+
+            _typeName_schema[schema.TypeName] = schema;
+            _typeName_isClusterSchema[schema.TypeName] = isClusterSchema;
+        }
+
+        private bool GetIsClusterSchema(Type serializedType, string typeName)
```

Comment:
I think the prefix `Get` on the method name is unnecessary but no harm.

Reply:
Makes sense, will rename